cloudpathlib/cloudpath.py

```diff
@@ -866,3 +892,81 @@ def _resolve(path: PurePosixPath) -> str:
         newpath = newpath + sep + name
 
     return newpath or sep
+
+
+# These objects are used to wrap CloudPaths in a context where we can use
+# the python pathlib implementations for `glob` and `rglob`, which depend
+# on the Selector created by the `_make_selector` method being passed
+# an object like the below when `select_from` is called. We implement these methods
+# in a simple wrapper to use the same glob recursion and pattern logic without
+# rolling our own.
+#
+# Designed to be compatible when used by these selector implementations from pathlib:
+# https://github.com/python/cpython/blob/3.10/Lib/pathlib.py#L385-L500
+class _CloudPathSelectableAccessor:
+    def __init__(self, scandir_func):
+        self.scandir = scandir_func
+
+
+class _CloudPathSelectable:
+    def __init__(
+        self, relative_cloud_path: PurePosixPath, children: List, is_dir: bool, exists: bool = True
```

Comment:
```suggestion
        self, relative_cloud_path: PurePosixPath, children: Dict[PurePosixPath, bool], is_dir: bool, exists: bool
```

- Fix type hint for `children`
- I think we don't need the default on `exists`. I think we always specify, and that's probably safer anyways.